metapool/metapool.py

```diff
@@ -2075,3 +2077,43 @@ def validate_plate_df(
             "The following sample names are duplicated %s"
             % ", ".join(sorted(duplicated_samples))
         )
+
+
+def generate_override_cycles_value(runinfo_xml_path, adapter_length):
+    if adapter_length < 0:
+        raise ValueError("Adapter-length cannot be less than zero")
+
+    tree = ET.parse(runinfo_xml_path)
+    reads = tree.getroot().findall('.Run/Reads/Read')
+    results = []
+    for read in reads:
+        result = {'number': int(read.get('Number')),
+                  'num_cycles': int(read.get('NumCycles'))}
+        result['is_indexed'] = True if read.get('IsIndexedRead') == 'Y' \
+            else False
+        results.append(result)
+
+    if len(results) == 0:
+        raise ValueError("Reads information could not be found in "
+                         f"'{runinfo_xml_path}'")
+
+    codes = []
+    # results should be in sorted order in XML file but this ensures that
+    # we process the elements in the correct order.
+    for read in sorted(results, key=itemgetter('number')):
+        if read['is_indexed'] is True:
+            code = f"I{adapter_length}"
+            truncated = read['num_cycles'] - adapter_length
+            if truncated < 0:
+                raise ValueError(f"num_cycles '{read['num_cycles']}' appears "
+                                 "to be less than adapter-length "
+                                 f"'{adapter_length}'")
+            elif truncated > 0:
+                code += f"N{truncated}"
+            # if truncated == 0 then a truncate code does not to be appended.
+        else:
+            code = f"Y{read['num_cycles']}"
```

Comment:
The string literal 'num_cycles' appears six times in this function.  I feel strongly that we need to move away from repeating strings this way in the codebase and instead set these repeated strings to constants and use those.

Reply:
Me too! Thanks for the reminder! :)